### DISCUSSION TITLE
Looking for help with debugging

### DISCUSSION THREAD

**Discussion by AbraxasDMG** [Open-Shell/Open-Shell-Menu#1661] [General]
Hi everyone.

I'd like to learn a bit about the win11-introduced changes in start menu button and windows hotkey functioning. I tried to do so by reading the source code but couldn't figure out what I'm interested in.

So, I downloaded the source code, opened it in VS 2022 community edition, made StartMenu the startu-up project and tried to debug it. I'm either doing something wrong or the projects had been set up in some special way (or the way the dll gets manipulated is causing that) because no breakpoints are hit in the StartMenuDLL project.

I'm not an experienced developer but with some other projects I have, I've been able to break on the breakpoints in the source code of the DLL that the main exe is using.

Could someone help/advise how can I instruct debugger to break on InitStartMenuDLL procedure within StartMenuDLL project?
Thanks a million in advance!

**Comment by ge0rdi** [comment]
Hello.

The thing is that `StartMenuDLL.dll` is running inside `explorer.exe` process as it needs to hook/subclass some things.
Thus you need to attach to `explorer.exe` and then breakpoints will start to work.

Here is how I'm doing it:

- Set `StartMenu` project as default (this should be like that when you clone repository)
- Select `Debug` `x64` configuration
- Exit `Open-Shell` (if running already)
- Start without debugging `Ctrl+F5`
This will compile all required projects and run `StartMenu.exe` so that `StartMenuDLL.dll` will get injected to explorer
- Attach to `explorer.exe` (using `Debug` -> `Attach to process`)
- You should be hitting breakpoints now

You can do this locally but keep in mind that you won't be able to use Explorer (including taskbar) when you stop on some breakpoint. So it may be cumbersome sometimes.
I'd recommend to use VM and remote debugging (but that's a different story).

**Reply by AbraxasDMG** [reply]
Thank you very much!
That explains a lot.
